plugins/module_utils/network/ios/facts/legacy/base.py

```diff
@@ -164,6 +168,37 @@ def parse_filesystems_info(self, data):
                 facts[fs]["spacefree_kb"] = int(match.group(2)) / 1024
         return facts
 
+    def parse_cpu_utilization(self, data):
+        facts = {}
+        for line in data.split("\n"):
+            regex_cpu_utilization = re.compile(
+                r"""
+                (^Core\s(?P<core>\d+)?:)?
+                (^|\s)CPU\sutilization\sfor\sfive\sseconds:
+                (\s(?P<f_sec>\d+)?%)?
+                (\s(?P<f_se_nom>\d+)%/(?P<f_s_denom>\d+)%\)?)?
+                ;\sone\sminute:\s(?P<a_min>\d+)?%
+                ;\sfive\sminutes:\s(?P<f_min>\d+)?%
+                """,
+                re.VERBOSE,
+            )
```

Comment:
You can just compile the regex once instead of recompiling it on every iteration of the loop:

```suggestion
        regex_cpu_utilization = re.compile(
            r"""
            (^Core\s(?P<core>\d+)?:)?
            (^|\s)CPU\sutilization\sfor\sfive\sseconds:
            (\s(?P<f_sec>\d+)?%)?
            (\s(?P<f_se_nom>\d+)%/(?P<f_s_denom>\d+)%\)?)?
            ;\sone\sminute:\s(?P<a_min>\d+)?%
            ;\sfive\sminutes:\s(?P<f_min>\d+)?%
            """,
            re.VERBOSE,
        )
        for line in data.split("\n"):
```